examples/plug-and-play/demo_PnP_custom_optim.py

```diff
@@ -0,0 +1,226 @@
+dataset_name = "set3c"
+img_size = 256 if torch.cuda.is_available() else 64
+url = "https://mycore.core-cloud.net/index.php/s/9EzDqcJxQUJKYul/download?path=%2Fdatasets&files=barbara.jpeg"
+x = load_image(
+    url=url, img_size=img_size, grayscale=True, resize_mode="resize", device=device
+)
+operation = "single_pixel"
+
+
+# %%
+# Generate a dataset of blurred images and load it.
+# --------------------------------------------------------------------------------
+# We use the BlurFFT class from the physics module to generate a dataset of blurred images.
+
+
+noise_level_img = 0.03  # Gaussian Noise standard deviation for the degradation
+n_channels = 1  # 3 for color images, 1 for gray-scale images
+physics = dinv.physics.SinglePixelCamera(
+    m=100,
+    img_shape=(1, 64, 64),
+    noise_model=dinv.physics.GaussianNoise(sigma=noise_level_img),
+)
+
+# Use parallel dataloader if using a GPU to fasten training,
+# otherwise, as all computes are on CPU, use synchronous data loading.
+num_workers = 4 if torch.cuda.is_available() else 0
+
+
+# %%
+# Set up the PnP algorithm to solve the inverse problem.
+# --------------------------------------------------------------------------------
+# We use the Proximal Gradient Descent optimization algoritm.
+# The algorithm alternates between a denoising step and a gradient descent step.
+# The denoising step is performed by a DNCNN pretrained denoiser :class:`deepinv.models.dncnn`.
+
+# Logging parameters
+verbose = True
+plot_metrics = True  # compute performance and convergence metrics along the algorithm, curved saved in RESULTS_DIR
+
+# Set up the PnP algorithm parameters : the `stepsize`, `g_param` the noise level of the denoiser and `lambda` the regularization parameter. The following parameters are chosen arbitrarily.
+params_algo = {"stepsize": 1.0, "g_param": noise_level_img, "lambda": 0.1}
+max_iter = 200
+early_stop = True  # stop the algorithm when convergence is reached
+
+# Select the data fidelity term
+data_fidelity = L2()
+
+# Specify the denoising prior
+denoiser = DnCNN(
+    in_channels=n_channels,
+    out_channels=n_channels,
+    pretrained="download",
+    train=False,
+    device=device,
+)
+prior = PnP(denoiser=denoiser)
+
+# instantiate the algorithm class to solve the IP problem.
+algo = CVIteration(data_fidelity=data_fidelity, F_fn=None, has_cost=False)
+model = optim_builder(
+    iteration=algo,
+    prior=prior,
+    data_fidelity=data_fidelity,
+    early_stop=early_stop,
+    max_iter=max_iter,
+    verbose=verbose,
+    params_algo=params_algo,
+    return_metrics=plot_metrics,
+)
+
+# %%
+# Evaluate the model on the problem and plot the results.
+# --------------------------------------------------------------------
+
+y = physics(x)
+x_lin = physics.A_adjoint(y)
+
+# run the model on the problem. When `return_metrics` is set to True, the model requires the ground-truth clean image ``x_gt`` and returns the output and the metrics computed along the iterations.
```

Comment:
is it possible to have return_metric = True and no gt?

Reply:
actually yes

Reply:
ok, maybe we can clarify in the comment